api/ocm/extensions/repositories/genericocireg/componentversion.go

```diff
@@ -259,32 +267,45 @@ func (c *ComponentVersionContainer) Update() (bool, error) {
 	return false, nil
 }
 
-func (c *ComponentVersionContainer) evalLayer(s compdesc.AccessSpec) (compdesc.AccessSpec, int, error) {
-	var d *artdesc.Descriptor
+func (c *ComponentVersionContainer) evalLayer(s compdesc.AccessSpec) (compdesc.AccessSpec, []int, error) {
+	var (
+		d         *artdesc.Descriptor
+		layernums []int
+	)
 
 	spec, err := c.GetContext().AccessSpecForSpec(s)
 	if err != nil {
-		return s, 0, err
+		return s, nil, err
 	}
 	if a, ok := spec.(*localblob.AccessSpec); ok {
 		if ok, _ := artdesc.IsDigest(a.LocalReference); !ok {
-			return s, 0, errors.ErrInvalid("digest", a.LocalReference)
+			return s, nil, errors.ErrInvalid("digest", a.LocalReference)
 		}
-		d = &artdesc.Descriptor{Digest: digest.Digest(a.LocalReference), MediaType: a.GetMimeType()}
-	}
-	if d != nil {
-		// find layer
-		layers := c.manifest.GetDescriptor().Layers
-		maxLen := len(layers) - 1
-		for i := range layers {
-			l := layers[len(layers)-1-i]
-			if i < maxLen && l.Digest == d.Digest && (d.Digest == "" || d.Digest == l.Digest) {
-				return s, len(layers) - 1 - i, nil
+		refs := strings.Split(a.LocalReference, ",")
+		media := a.GetMimeType()
+		if len(refs) > 1 {
+			media = mime.MIME_OCTET
+		}
+		for _, ref := range refs {
+			d = &artdesc.Descriptor{Digest: digest.Digest(strings.TrimSpace(ref)), MediaType: media}
+			// find layer
+			layers := c.manifest.GetDescriptor().Layers
+			maxLen := len(layers) - 1
+			found := false
+			for i := range layers {
+				l := layers[len(layers)-1-i]
+				if i < maxLen && l.Digest == d.Digest && (d.Digest == "" || d.Digest == l.Digest) {
```

Comment:
The second `&& (d.Digest == "" || d.Digest == l.Digest)` is redundant.

Reply:
yes